pkg/terraform/exec/plugins/vsphereprivate/resource_vsphereprivate_import_ova.go

```diff
@@ -375,6 +423,21 @@ func resourceVSpherePrivateImportOvaCreate(d *schema.ResourceData, meta interfac
 	}
 	log.Printf("[DEBUG] %s: mark as template", vm.Name())
 
+	task, err := vm.UpgradeVM(ctx, importOvaParams.HardwareVersion)
```

Comment:
do we need more context here? like a comment or something in the commit message? I'm not sure what this is doing or why.